interview-questions.md

```diff
@@ -7,62 +7,146 @@ Then, research the question to expand on your answer. Even if you feel you have
+                they have different competitors and different purposes. Both Git and Github are used to complement each other and make managing code
+                easier.
+    References: https://www.simplilearn.com/tutorials/git-tutorial/git-vs-github#git_and_github_operate_completely_differently
+                https://www.geeksforgeeks.org/difference-between-git-and-github/
+                
+                
 
 
 2. Which JavaScript operators will return a Boolean value?
 
-  Your answer:
+  Your answer: 
+               JavaScript logical operators , && (and) || (or) used in tandem with conditionals return the 
+               value true or false. 
 
   Researched answer:
+                In addition , equality operators will also return a Boolean value. Loosely == or Strictly === comparing
+                values will return true/false. We can also combine that with the logical not ! operator to reverse the 
+                turn truthy to falsy and vice versa. I don't have very much exposure with this concept yet but we can use 
+                a double not !! to "to explicitly force the conversion of any value to the corresponding boolean primitive." -MDN
+    References: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_NOT
+                https://github.com/learn-academy-2022-alpha/Syllabus/blob/main/javascript/intro.md
 
 
 
 3. What is an index? What is the difference between index and value?
 
-  Your answer:
+  Your answer: Index would be the place of a specified item, while value is what that specified item is. For example,
+               finding what place of a string a certain letter is or what place an element of an array occurs would be 
+               the index, while the letter or specific word would be the value. 
 
   Researched answer:
+               We can attach [] to the end of a string to find the placement of a particular character contained in that string.
+               Using the index method on a array will return the element located at the specified index. Just like strings
+               arrays are zero-indexed . Meaning the first letter or element would be 0 , and so forth. If the specified char/element index is not found
+               then a -1 will be the resulting output.Values are generally contain properties that define variables. They are often compared, assigned 
+               and altered to suit our coding needs. In order to handle the massive amount of information and date we need to manipulate
+               we can look at values as manageable bits of information.
+   References: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array
+               https://eloquentjavascript.net/01_values.html 
+                
 
 
 
 4. What is iteration?
 
-  Your answer:
+  Your answer: Iteration in programming is where a specified value is checked and evaluated against every occurrence
```

Comment:
Yes, the iteration will continue until a specified amount of time is met, or a condition is met